### PR TITLE
test,ci,feat: (1) Use poetry for pkg management. (2) Improve CI. (3) Re-gen protos for v0.12.1-alpha of nibiru

### LINKED ISSUE #45: Move package management over to `poetry`
Poetry is a tool for dependency management and packaging in Python. It allows you to declare the libraries your project depends on and it will manage (install/update) them for you.
 
- usage instructions: https://python-poetry.org/docs/basic-usage/
- installation instructions: https://python-poetry.org/docs/#osx--linux--bashonwindows-install-instructions

`poetry` edits dependencies directly in the pyproject.toml and works well for versioning and publishing.

### PR DESCRIPTION
**Pull request by Unique-Divine**
### For the reviewer 

You'll need the environment variables listed in the [repo configs notion page](https://www.notion.so/nibiru/Resources-and-Repo-Configs-b31aa8074a2b419d80b0c946ed5efab0).

The substance of the PR lies within 5 files: 

- `nibiru/network.py`: Created `devnet
- `pyproject.toml` (and poetry.lock): Made deterministic build process and used poetry for dependency resolution. 
- `github/workflows/.pytests.yml`: Verified correctness of the new setup with CI
- `tests/chain_info_test.py`: Cleaned up tests and added a vpool query to the suite
- `scipts/protocgen.sh`: Improved this script to run whether or not the proto directory exists. Added docs to go with it to the README.

### Summary - Commit Log

- ci: Use poetry for the test workflow. Handle concurrency. Add cacheing.
- fix: make it so that make proto-gen runs even if the proto dir already exists
- feat(network): add devnet to the Network class
- test(chain_info_test): test_query_vpool_reserve_assets
- docs(README): Instructions on generating types with protobuf
- proto: regenerare protos wth binary v0.12.1-alpha
- feat: use poetry instead of pipenv b/c it's objectively better. Closes #45

### CODE REVIEW

File: nibiru/clients/perp.py [hunk 1]
```diff
@@ -36,6 +36,7 @@ def params(self):
         output = MessageToDict(proto_output)["params"]
 
         sdk_dec_fields = [
+            # "maintenanceMarginRatio",
```
Comment (matthiasmatt) [inline nibiru/clients/perp.py:39]: This is not in the perp query anymore, but moved to the vpool queries. We should remove this commented line

**Review by matthiasmatt — CHANGES_REQUESTED** [review]

**Review by matthiasmatt — DISMISSED** [review]

**Review by matthiasmatt — APPROVED** [review]

File: scripts/protocgen.sh [hunk 2]
```diff
@@ -14,17 +14,31 @@ protoc_gen_gocosmos() {
   go get github.com/cosmos/cosmos-sdk@v0.45.6 2>/dev/null
 }
 
-# refresh existing proto files
-rm -rf nibiru/proto proto/
-mkdir -p nibiru/proto/
-cp -r ../nibiru/proto/ proto/
+echo "refresh existing proto files"
+if [ $(basename $(pwd)) = nibiru-py ]
```
[the remark this answers is not in the document]
Comment (Unique-Divine) [reply scripts/protocgen.sh:18]: I'm using it as an indirect form of documentation. It's a bit hard to follow this script because the the `nibiru` repo is assumed to be at `../nibiru`, while the `nibiru` package is `nibiru` and the script actually `cd`s around. 
My intention here was to hint at where the execution was taking place. 

We should probably add an error message for when the nibiru repo isn't found as well to make the dev exp. a little better.

File: .github/workflows/pytests.yml [hunk 3]
```diff
@@ -1,39 +1,78 @@
...
-      uses: actions/cache@v1
-      with:
-        path: ~/.local/share/virtualenvs
-        key: ${{ runner.os }}-pipenv-${{ hashFiles('**/Pipfile.lock') }}
 
-    - name: Install dependencies
-      if: steps.cache-pipenv.outputs.cache-hit != 'true'
-      run: |
-        pipenv install --dev
+    # - name: Install pipenv with pip
```
[the remark this answers is not in the document]
Comment (Unique-Divine) [reply .github/workflows/pytests.yml:63]: done db76f5726d7907cb0488868b4b60ba769e7c1183

File: .github/workflows/pytests.yml [hunk 4]
```diff
@@ -1,39 +1,78 @@
 name: Run tests
 
-on: [push]
+on: 
+  pull_request:
+    branches: ["master", "release/*"]
```
[the remark this answers is not in the document]
Comment (Unique-Divine) [reply .github/workflows/pytests.yml]: done db76f57

File: tests/chain_info_test.py [hunk 5]
```diff
@@ -9,64 +8,57 @@
...
-        fee_denom='unibi',
-        env="local",
-    )
+def devnet() -> nibiru.Network:
+    return nibiru.Network.devnet()
 
 
 @pytest.fixture
-def val_node(network: nibiru.Network) -> nibiru.Sdk:
+def val_node(devnet: nibiru.Network) -> nibiru.Sdk:
```
[the remark this answers is not in the document]
Comment (Unique-Divine) [reply tests/chain_info_test.py]: done 2e843d4
